src/task_queue/tasks.py

```diff
@@ -1,63 +1,62 @@
 def run_some_task(self, task_id):
-    
-    lock_id = f'task_lock_{task_id}'
-    cached_lock = cache.get(lock_id)
-    
-    if cached_lock:
-        return
-    
-    cache.set(lock_id, True)
-    
-    try:
-        today = datetime.date.today()
-        should_run = False
-        task = None
-        with transaction.atomic():
-            task = Task.objects.select_for_update().get(id=task_id)  # locks task instance until transaction is committed
-            if is_time_to_run(task):
-                task.last_run = today
-                task.save()
-                should_run = True
-
-        if task and should_run:
-            func = FUNCTIONS.get(task.job_name)
-            if func:
-                task.status = "SCHEDULED"
-                task.save()
-                try:
-                    res = func(task)
-                    task.status = "SUCCEEDED" if res else "FAILED"
-                    if IS_PROD:
-                        send_slack_message(SLACK_SUPER_ADMINS_WEBHOOK_URL, {
-                            "text": f"({ENV}): Task '{task.job_name}' was run on {today}. Status: {task.status} "})
-                except Exception as e:
-                    task.status = "FAILED"
-                    if IS_PROD:
-                        send_slack_message(SLACK_SUPER_ADMINS_WEBHOOK_URL, {
-                            "text": f"({ENV}): Task '{task.job_name}' was run on {today}. Status: {task.status}. Reason: {str(e)} "})
-            else:
-                task.status = "FAILED"
-                if IS_PROD:
-                    send_slack_message(SLACK_SUPER_ADMINS_WEBHOOK_URL, {
-                        "text": f"({ENV}): Task '{task.job_name}' was run on {today}. Status: {task.status}. Reason: No function found for job_name '{task.job_name}'"})
-            task.save()
-    except Exception as e:
-        if IS_PROD:
-            send_slack_message(SLACK_SUPER_ADMINS_WEBHOOK_URL, {
-                "text": f"({ENV}): Task '{task.job_name}' was run on {today}. Status: FAILED. Reason: {str(e)} "})
-    finally:
-        cache.delete(lock_id)
\ No newline at end of file
+	try:
+		today = datetime.date.today()
+		should_run = False
+		task = None
+		
+		with transaction.atomic():
+			task = Task.objects.select_for_update().get(id=task_id)
+			
+			if is_time_to_run(task):
+				task.last_run = today
+				task.save()
+				should_run = True
+		
+		if task and should_run:
+			func = FUNCTIONS.get(task.job_name)
+			if func:
+				try:
+					res = func(task)
+					task.status = "SUCCEEDED" if res else "FAILED"
+					log_status(task.status, task.job_name, "Task ran successfully" if res else "Task failed")
+				
+				except Exception as e:
+					task.status = "FAILED"
+					log_status(task.status, task.job_name, today, str(e))
+			else:
+				task.status = "FAILED"
+				log_status(task.status, task.job_name, "Task not found in FUNCTIONS")
+			task.save()
+			if task.frequency == "ONE_OFF":
```

Comment:
PLEASE - don't archive ONE_OFF tasks.  Leave them in the list so they can run again.

Reply:
Ok